packages/dds/tree/src/shared-tree/treeCheckout.ts

```diff
@@ -604,6 +664,19 @@ export class TreeCheckout implements ITreeCheckoutFork {
 		this.revertibles.delete(revertible);
 	}
 
+	private disposeRevertibleRoots(revision: RevisionTag): void {
+		const roots = this.removedRoots.getRoots(revision);
```

Comment:
Doesn't this run the risk of deleting data is still relevant to other changes? If it is indeed an issue, please try to write a test to catch the issue this before fixing it.